config/accounts_test.go

```diff
@@ -246,11 +246,14 @@ func TestAccountIntegrationGetByIntegrationType(t *testing.T) {
 }
 
 func TestPurposeEnforced(t *testing.T) {
+	True := true
+	False := false
+
 	tests := []struct {
 		description          string
 		givePurposeConfigNil bool
-		givePurpose1Enforced string
-		givePurpose2Enforced string
+		givePurpose1Enforced *bool
+		givePurpose2Enforced *bool
```

Comment:
Could you explain why the type change here?

Reply:
Sure, the function here under test is the account config method `PurposeEnforced` which indicates whether a given purpose is enforced. That function looks at whether `EnforcePurpose` is set on the account config. `EnforcePurpose` used to be a string value and now, as of this PR, it's a bool so these two values have changed from string to bool since they are used to setup the config for the test. They are also pointers just like the `EnforcePurpose` field on the `AccountGDPRPurpose` object because it allows us to detect presence. Before we were able to detect presence when it was just a string (no pointer) by looking for an empty string since the only valid values were non-empty values "full" and "no".